src/main/java/org/apache/sysds/runtime/privacy/finegrained/FineGrainedPrivacyList.java

```diff
@@ -35,11 +35,61 @@
 
 	private ArrayList<Map.Entry<DataRange, PrivacyLevel>> constraintCollection = new ArrayList<>();
```

Comment:
why is it an Array List of Map Entries. 

You can use Immutable Pairs, if you want to stick to Lists.
or maybe use a Hash Map instead of array list?

Have you measured how it performs, since i assume that what you want it to search through ranges and get the privacy associated with that specific data?

Reply:
This is not really a part of the PR. I will look at other ways to implement the FineGrainedPrivacy interface when it becomes relevant. 
There are several classes that represent pairs that could be relevant to use. I picked Map.Entry since it was a part of java.util and supported the operations I needed.

Reply:
sounds reasonable, but maybe add a JIRA task to remember it later